Jellyfin.Server/Migrations/Routines/MoveTrickplayFiles.cs

```diff
@@ -42,26 +50,50 @@ public MoveTrickplayFiles(ITrickplayManager trickplayManager, IFileSystem fileSy
     /// <inheritdoc />
     public void Perform()
     {
-        var trickplayItems = _trickplayManager.GetTrickplayItemsAsync().GetAwaiter().GetResult();
-        foreach (var itemId in trickplayItems)
+        const int Limit = 100;
+        int itemCount = 0, offset = 0, previousCount;
+
+        var sw = Stopwatch.StartNew();
+        var trickplayQuery = new InternalItemsQuery
         {
-            var resolutions = _trickplayManager.GetTrickplayResolutions(itemId).GetAwaiter().GetResult();
-            var item = _libraryManager.GetItemById(itemId);
-            if (item is null)
-            {
-                continue;
-            }
+            MediaTypes = [MediaType.Video],
+            SourceTypes = [SourceType.Library],
+            IsVirtualItem = false,
+            IsFolder = false
+        };
 
-            foreach (var resolution in resolutions)
+        do
+        {
+            var trickplayInfos = _trickplayManager.GetTrickplayItemsAsync(Limit, offset).GetAwaiter().GetResult();
+            previousCount = trickplayInfos.Count;
+            offset += Limit;
+
+            trickplayQuery.ItemIds = trickplayInfos.Select(i => i.ItemId).Distinct().ToArray();
+            var items = _libraryManager.GetItemList(trickplayQuery);
+            foreach (var trickplayInfo in trickplayInfos)
             {
-                var oldPath = GetOldTrickplayDirectory(item, resolution.Key);
-                var newPath = _trickplayManager.GetTrickplayDirectory(item, resolution.Value.TileWidth, resolution.Value.TileHeight, resolution.Value.Width, false);
+                var item = items.OfType<Video>().FirstOrDefault(i => i.Id.Equals(trickplayInfo.ItemId));
+                if (item is null)
+                {
+                    continue;
+                }
+
+                if (++itemCount % 1_000 == 0)
+                {
+                    _logger.LogInformation("Moved {Count} items in {Time}", itemCount, sw.Elapsed);
+                }
+
+                var oldPath = GetOldTrickplayDirectory(item, trickplayInfo.Width);
+                var newPath = _trickplayManager.GetTrickplayDirectory(item, trickplayInfo.TileWidth, trickplayInfo.TileHeight, trickplayInfo.Width, false);
                 if (_fileSystem.DirectoryExists(oldPath))
                 {
                     _fileSystem.MoveDirectory(oldPath, newPath);
                 }
             }
-        }
+        } while (previousCount != 0);
+
+        _logger.LogInformation("Moved {Count} items in {Time}", itemCount, sw.Elapsed);
+        throw new FfmpegException("Don't save me!");
```

Comment:
I guess this is debugging code?

Reply:
:cold_sweat: no idea what you're talking about